Core/Src/monitor.c

```diff
@@ -68,4 +68,22 @@ void vTempMonitor(void *pv_params)
 		/* Yield to other tasks */
 		osDelayUntil(temp_sensor_sample_delay);
 	}
-}
\ No newline at end of file
+}
+
+osThreadId_t watchdog_monitor_handle;
+const osThreadAttr_t watchdog_monitor_attributes = {
+	.name = "WatchdogMonitor",
+	.stack_size = 128 * 4,
+	.priority = (osPriority_t) osPriorityLow,
+};
+
+void vWatchdogMonitor(void *pv_params)
+{
+	GPIO_TypeDef *gpio;
+	gpio = (GPIO_TypeDef *)pv_params;
+
+	for(;;) {
+		/* Pets Watchdog */
+		HAL_GPIO_WritePin(gpio, GPIO_PIN_15, GPIO_PIN_SET);
```

Comment:
Okay so this will just set the watchdog, we need to set it low then high again. This just constantly sets it high which wouldn't be any change in status that the watchdog IC can detect. I think we need to set it low, wait a certain number of milliseconds or nanoseconds, and then set it high. So it might be:
```
init to high;

while 1:
writeHigh;
delay(5ms);
writeLow;
yieldToScheduler();

Reply:
yeah i thought that was what was happening but didnt know if stm just automatically wrote low after a set time.